actors/test/miner_scenario_test.go

```diff
@@ -0,0 +1,315 @@
+	vm "github.com/filecoin-project/specs-actors/support/vm"
+
+	"github.com/filecoin-project/go-bitfield"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCommitPoStFlow(t *testing.T) {
+	ctx := context.Background()
+	v := vm.NewVMWithSingletons(ctx, t)
+	addrs := vm.CreateAccounts(ctx, t, v, 1, big.Mul(big.NewInt(10_000), big.NewInt(1e18)), 93837778)
+
+	minerBalance := big.Mul(big.NewInt(10_000), vm.FIL)
+	sectorNumber := abi.SectorNumber(100)
+	sealedCid := tutil.MakeCID("100", &miner.SealedCIDPrefix)
+	sealProof := abi.RegisteredSealProof_StackedDrg32GiBV1
+
+	// create miner
+	params := power.CreateMinerParams{
+		Owner:         addrs[0],
+		Worker:        addrs[0],
+		SealProofType: sealProof,
+		Peer:          abi.PeerID("not really a peer id"),
+	}
+	ret, code := v.ApplyMessage(addrs[0], builtin.StoragePowerActorAddr, minerBalance, builtin.MethodsPower.CreateMiner, &params)
+	require.Equal(t, exitcode.Ok, code)
+
+	minerAddrs, ok := ret.(*power.CreateMinerReturn)
+	require.True(t, ok)
+
+	// advance vm so we can have seal randomness epoch in the past
+	v, err := v.WithEpoch(200)
+	require.NoError(t, err)
+
+	//
+	// precommit sector
+	//
+
+	preCommitParams := miner.SectorPreCommitInfo{
+		SealProof:     sealProof,
+		SectorNumber:  sectorNumber,
+		SealedCID:     sealedCid,
+		SealRandEpoch: v.GetEpoch() - 1,
+		DealIDs:       nil,
+		Expiration:    v.GetEpoch() + miner.MinSectorExpiration + miner.MaxSealDuration[sealProof] + 100,
+	}
+	_, code = v.ApplyMessage(addrs[0], minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.PreCommitSector, &preCommitParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	// assert successful precommit invocation
+	vm.ExpectInvocation{
+		// Original send to storage power actor
+		To:     minerAddrs.IDAddress,
+		Method: builtin.MethodsMiner.PreCommitSector,
+		Params: vm.ExpectObject(&preCommitParams),
+		SubInvocations: []vm.ExpectInvocation{
+			{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.ThisEpochReward},
+			{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.CurrentTotalPower},
+			{To: builtin.StorageMarketActorAddr, Method: builtin.MethodsMarket.VerifyDealsForActivation}},
+	}.Matches(t, v.Invocations()[0])
+
+	// find information about precommited sector
+	var minerState miner.State
+	err = v.GetState(minerAddrs.IDAddress, &minerState)
+	require.NoError(t, err)
+
+	precommit, found, err := minerState.GetPrecommittedSector(v.Store(), sectorNumber)
+	require.NoError(t, err)
+	require.True(t, found)
+
+	// advance time to max seal duration
+	proveTime := v.GetEpoch() + miner.MaxSealDuration[sealProof]
+	v, dlInfo := vm.AdvanceTillEpoch(t, v, minerAddrs.IDAddress, proveTime)
+
+	//
+	// overdue precommit
+	//
+
+	t.Run("missed prove commit results in precommit expiry", func(t *testing.T) {
+		// advanced one more deadline so precommit is late
+		tv, _ := v.WithEpoch(dlInfo.Close)
+		require.NoError(t, err)
```

Comment:
`err` is not set as expected here.

Higher level, I think we should strive to remove error returns from the test helpers. fail the test or panic instead so the test code can be tighter.